frontend/src/components/car-list-item/car-list-item.tsx

```diff
@@ -1,41 +1,79 @@
 
-import { ComplectationDetailsType } from '@autoline/shared/common/types/types';
+import {
+  ComplectationDetailsType,
+  WishlistInput,
+} from '@autoline/shared/common/types/types';
+import { AppRoute } from '@common/enums/enums';
 import { CarListItemProps } from '@common/types/types';
 import { SliderNavButton } from '@components/car-list-item/slider-nav-button/slider-nav-button';
 import { swiperParams } from '@components/car-list-item/swiper-params';
+import { HeartIcon } from '@components/common/icons/icons';
 import { Spinner } from '@components/common/spinner/spinner';
 import { CompleteSetTableCollapsed } from '@components/complete-set-table/complete-set-table-collapsed';
 import { formatPrice } from '@helpers/helpers';
 import { objectToQueryString } from '@helpers/object-to-query';
+import { useAppSelector } from '@hooks/hooks';
 import { Grid } from '@mui/material';
 import { uuid4 } from '@sentry/utils';
 import {
   useGetComplectationsQuery,
   useGetModelDetailsQuery,
 } from '@store/queries/cars';
+import {
+  useGetWishlistsLikedItemQuery,
+  useCreateWishlistMutation,
+  useDeleteWishlistMutation,
+} from '@store/queries/preferences/wishlist';
 import { clsx } from 'clsx';
 import { Swiper, SwiperSlide } from 'swiper/react';
 
 import styles from './styles.module.scss';
 
 const CarListItem: React.FC<CarListItemProps> = (props) => {
   const { model_id, complectations_id } = props;
+  const authToken = useAppSelector((state) => state.auth.token);
+  const navigate = useNavigate();
 
   const idParams = objectToQueryString({ 'id': complectations_id });
 
   const { data: complectations = [], isLoading: isComplectationsLoading } =
     useGetComplectationsQuery(idParams);
   const { data: model, isLoading: isModelLoading } =
     useGetModelDetailsQuery(model_id);
-
+  const { data: isLikedModel } = useGetWishlistsLikedItemQuery({
+    modelId: model_id,
+  });
   const modelName = `${model?.brandName} ${model?.modelName}`;
   const [modelPrice, setModelPrice] = useState<string>();
   const [modelOptions, setModelOptions] = useState<{ name: string }[]>();
   const [modelEngine, setModelEngine] = useState<string>();
   const [modelEnginePower, setModelEnginePower] = useState<string>();
   const [modelFuelType, setModelFuelType] = useState<string>();
   const [modelTransmission, setModelTransmission] = useState<string>();
+  const [createWishlist] = useCreateWishlistMutation();
+  const [deleteWishlist] = useDeleteWishlistMutation();
+
+  const handleCreateWishlist = async (data: WishlistInput): Promise<void> => {
+    await createWishlist(data);
+  };
+
+  const handleDeleteWishlist = async (data: WishlistInput): Promise<void> => {
+    await deleteWishlist(data);
+  };
+
+  const handleLikeClick = (event: React.MouseEvent): void => {
+    event.stopPropagation();
+    const data: WishlistInput = { modelId: model_id };
+
+    if (!authToken) {
+      navigate(AppRoute.SIGN_IN);
+      return;
+    }
```

Comment:
This should be handled in some shared place. Like a RTK middleware. Because we have the same exact code for the likes on the cards